### PR TITLE
SS-7926: adjusted name filtering

### PR DESCRIPTION
**Pull request by MajorMeerkatThe3rd**
https://shapediver.atlassian.net/browse/SS-7926

@snabela please review this new logic!

### CODE REVIEW

File: hooks/shapediver/viewer/interaction/useNodeInteractionData.ts [hunk 1]
```diff
@@ -79,7 +50,7 @@ export function useNodeInteractionData(sessionId: string, outputIdOrName: string
 		 * @param node 
 		 * @param interactionDataSettings 
 		 */
-		const addInteractionData = (node: ITreeNode, interactionDataSettings: { select?: boolean, hover?: boolean, drag?: boolean}) => {
+		const addInteractionData = (node: ITreeNode, interactionDataSettings: { select?: boolean, hover?: boolean, drag?: boolean }) => {
```
Comment (snabela) [inline hooks/shapediver/viewer/interaction/useNodeInteractionData.ts:53]: @MajorMeerkatThe3rd this looks like a helper function for managing data in `nodesWithInteractionData`, seems like we can move it outside of here to the global scope of this file (where `nodesWithInteractionData` is defined).

File: hooks/shapediver/viewer/interaction/utils/patternUtils.ts [hunk 2]
```diff
@@ -21,6 +21,47 @@ import { IOutputApi, ITreeNode, OutputApiData } from "@shapediver/viewer";
...
+ * @param result The result array. 
+ */
+export const gatherNodesForPattern = (node: ITreeNode, pattern: string[], count: number, result: ITreeNode[] = []): void => {
+	// if there is no original name or the node name is in the black list, ignore the node
+	if (!node.originalName || nodeNameBlackList.includes(node.originalName)) {
+		if (count !== 0) return;
+
+		for (const child of node.children) {
+			gatherNodesForPattern(child, pattern, 0, result);
+		}
```
Comment (snabela) [inline hooks/shapediver/viewer/interaction/utils/patternUtils.ts:51]: @MajorMeerkatThe3rd this special logic for count === 0 is not clear to me, please explain it in the code documentation

File: hooks/shapediver/viewer/interaction/utils/patternUtils.ts [hunk 3]
```diff
@@ -21,6 +21,47 @@ import { IOutputApi, ITreeNode, OutputApiData } from "@shapediver/viewer";
...
+ * 
+ * For each pattern, the function will check if the node name matches the pattern.
+ * If the node name matches the pattern, the function will check the children of the node.
+ * Only if the node name matches the last pattern, the interaction data will be added.
+ * 
+ * @param node The node to check.
+ * @param pattern The pattern to check.
+ * @param count The current count of the pattern.
+ * @param result The result array. 
+ */
```
Comment (snabela) [inline hooks/shapediver/viewer/interaction/utils/patternUtils.ts:43]: @MajorMeerkatThe3rd let's review and improve this function documentation

**Comment by MajorMeerkatThe3rd** [comment]
@snabela do you have any other names to add to the `nodeNameBlackList`?

`const nodeNameBlackList = ["TransformZUpToYUp", "no_transformations"];`

**Comment by snabela** [comment]
> @snabela do you have any other names to add to the `nodeNameBlackList`?
> 
> `const nodeNameBlackList = ["TransformZUpToYUp", "no_transformations"];`

@MajorMeerkatThe3rd sorry I had forgotten to answer your question: I reviewed this and the list is complete.

**Pull request merged**